btrack/io/_localization.py

```diff
@@ -16,119 +18,184 @@
 logger = logging.getLogger(__name__)
 
 
-def _centroids_from_single_arr(
-    segmentation: Union[np.ndarray, Generator],
-    properties: Tuple[str],
-    frame: int,
-    intensity_image: Optional[np.ndarray] = None,
-    scale: Optional[Tuple[float]] = None,
-    *,
-    use_weighted_centroid: bool = False,
-    assign_class_ID: bool = False,
-) -> np.ndarray:
-    """Return the object centroids from a numpy array representing the
-    image data."""
+try:
+    from tqdm import tqdm
+except ImportError:
+    # this provides a dummy progress bar in case `tqdm` is not installed.
+    def tqdm(iterator, *args, **kwargs):
+        logger.info("Try installing ``tqdm`` for progress bar rendering.")
+        return iterator
```

Comment:
Why wouldn't `tqdm` be installed?

Reply:
I just left it optional, we could add to the requirements

Reply:
If we want it optional could add something similar to `optional-dependencies.docs`

Reply:
I've added it as a requirement. I think it could be useful in other parts of the codebase too